README.md

```diff
@@ -20,6 +20,58 @@ Supported features are:
 - `snapshot-restore`: Restore a snapshot
 - `snapshot-clean`: Delete all snapshots
 
+### Jest test database manager
+`TestDatabaseManager` is a helper with managing databases in Jest tests.
+The ultimate purpose is to make sure that all test suites are run in isolation.
+Data created in one tests shouldn't affect other tests.
+How it works is:
+- During Jest setup process, an empty base database is created. Then all the migration scripts are run on the base database.
+- For each test suite, a brand new database is created by cloning the base database. All tests in the suite are run against this database.
+  When the test suite finishes running, the database is dropped.
+- When all Jest tests finishes, the base database is dropped.
+
+To use it, please follow the below steps:
+
+- In Jest setup, initialise the test database manager
+```typescript
+export const testDbManager = new TestDatabaseManager({
+  typeOrmOptions: typeOrmConfig,
+  dbPrefix: 'db_prefix_',
+  adminUsername: 'admin',
+  adminPassword: 'password',
+  postgresDatabaseName: 'postgres',
+})
+
+// in jest.setup,ts
+export default async (): Promise<void> => {
+  config({ override: true })
+  await testDbManager.init()
+}
+```
+- In each test suite, create new database at the beginning and drop it at the end
+```typescript
```

Comment:
Would this be more reflective of real life and easier to understand at a glance to assign both the DataSource + dispose rather than just assigning the dispose function?

```suggestion
describe('example suite which uses isolated DB instance', () => {
  let dispose: AsyncDisposable
  let ds: DataSource

  beforeAll(async () => {
      const prepared = await testDbManager.prepareTestDatabaseSource()
      disposable = prepared.disposable
      dataSource = prepared.dataSource
  })
  afterAll(async () => {
    await dataSource.destroy()
    await dispose()
  })
})
```